hydromt/data_catalog/drivers/dataset/xarray_driver.py

```diff
@@ -130,18 +133,49 @@ def read(
                 )
         return ds
 
-    def write(self, path: StrPath, ds: xr.Dataset, **kwargs) -> str:
+    def write(
+        self,
+        path: Path | str,
+        data: xr.Dataset,
+        *,
+        write_kwargs: dict[str, Any] | None = None,
+    ) -> Path:
         """
-        Write the Dataset to a local file using zarr.
-
-        args:
+        Write an xarray Dataset to disk using the xarray I/O engine.
+
+        Supports writing to both Zarr and NetCDF formats. The file format is inferred
+        from the file extension. If the extension is not recognized, a ValueError is raised.
+
+        Parameters
+        ----------
+        path : Path | str
+            Destination path or URI where the Dataset will be written.
+            The file extension determines the output format:
+            `.zarr`, `.nc`, or `.netcdf`.
+        data : xr.Dataset
+            The Dataset to write to disk.
+        write_kwargs : dict[str, Any], optional
+            Additional keyword arguments passed to the xarray write function
+            (`Dataset.to_zarr` or `Dataset.to_netcdf`). Default is None.
+
+        Returns
+        -------
+        Path
+            The path where the dataset was written.
+
+        Raises
+        ------
+        ValueError
+            If the provided file extension is unsupported.
         """
-        ext = splitext(path)[-1]
+        if isinstance(path, str):
+            path = Path(path)
```

Comment:
```suggestion
        path = Path(path)
```

You can always do this, even when it's a Path.